pkg/schedule/labeler/labeler.go

```diff
@@ -272,9 +272,20 @@ func (l *RegionLabeler) Patch(patch LabelRulePatch) error {
 		}
 	}
 
+	// Handling duplicate entries in DeleteRules and SetRules
+	// Note: We maintain compatibility with the previous behavior, which is to process DeleteRules before SetRules
+	// If there are duplicate rules, we will prioritize SetRules
+	deleteRules := make(map[string]struct{})
+	for _, ruleID := range patch.DeleteRules {
```

Comment:
Iterate and build the set rules first could eliminate one traversal. We just need to skip the rules in set later while building the delete ruels.